src/main/java/fr/inria/coming/repairability/repairtools/Arja.java

```diff
@@ -0,0 +1,169 @@
+package fr.inria.coming.repairability.repairtools;
+
+import fr.inria.coming.changeminer.analyzer.instancedetector.ChangePatternInstance;
+import fr.inria.coming.changeminer.analyzer.patternspecification.ChangePatternSpecification;
+import fr.inria.coming.changeminer.entity.IRevision;
+import fr.inria.coming.changeminer.util.PatternXMLParser;
+import gumtree.spoon.diff.operations.*;
+import spoon.Launcher;
+import spoon.SpoonException;
+import spoon.pattern.Match;
+import spoon.pattern.Pattern;
+import spoon.pattern.PatternBuilder;
+import spoon.reflect.CtModel;
+import spoon.reflect.code.CtBinaryOperator;
+import spoon.reflect.code.CtConstructorCall;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.declaration.*;
+import spoon.reflect.path.CtRole;
+import spoon.reflect.reference.CtFieldReference;
+import spoon.reflect.reference.CtTypeReference;
+import spoon.reflect.visitor.filter.TypeFilter;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class Arja extends AbstractRepairTool {
```

Comment:
documentation to give the context, with pointers to the related papers

Reply:
documentation added.